### PR TITLE
Upload api signature

### PR DESCRIPTION
**Pull request by SteveRedka**

### CODE REVIEW

File: lib/uploadcare/client/upload_client.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      response.fmap { |files| { 'files': files.map { |fname, uuid| { original_filename: fname.to_s, uuid: uuid } } } }
+    end
+
+    private
+
+    def upload_params(store = false)
+      {
+        'UPLOADCARE_PUB_KEY': PUBLIC_KEY,
+        'UPLOADCARE_STORE': (store == true) ? '1' : '0',
+        'signature': Upload::SignatureGenerator.call
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/upload_client.rb]: this should be added only when it's required (most likely via config setting).

Comment (SteveRedka) [reply lib/uploadcare/client/upload_client.rb]: If you don't specify that you need it in Uploadcare settings, these fields are ignored

Comment (SteveRedka) [reply lib/uploadcare/client/upload_client.rb]: Ok, added that and made signing disabled by default

File: lib/uploadcare/service/upload/upload_params_generator.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'digest'
+
+module Uploadcare
+  module Upload
+    # This class generates body params for uploads
+    # https://uploadcare.com/docs/api_reference/upload/request_based/
+    class UploadParamsGenerator
+      def self.call(store = false)
```
Comment (dmitry-mukhin) [inline lib/uploadcare/service/upload/upload_params_generator.rb]:
```suggestion
      def self.call(store = 'auto')
```

File: lib/uploadcare/service/upload/signature_generator.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# This file generates signatures for protected uploads
+# https://uploadcare.com/docs/api_reference/upload/signed_uploads/
+
+require 'digest'
+
+module Uploadcare
+  module Upload
+    class SignatureGenerator
+      def self.call
+        expires_at = Time.now.to_i + 60 * 30
```
Comment (dmitry-mukhin) [inline lib/uploadcare/service/upload/signature_generator.rb]: ideally, we want signature lifetime to be configurable.
30 minutes is too much, anyway

**Pull request merged**